src/adapter/graphql/src/scalars/dataset_metadata.rs

```diff
@@ -0,0 +1,107 @@
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::convert::Into;
+
+use async_graphql::*;
+use kamu_core::utils::metadata_chain_comparator as comp;
+use opendatafabric as odf;
+use serde::Serialize;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// DatasetPushStatuses
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(SimpleObject, Clone, Debug)]
+pub struct DatasetPushStatus {
+    pub remote: DatasetRefRemote,
+    pub status: Option<CompareChainsResult>,
+}
+
+#[derive(SimpleObject, Clone, Debug)]
+pub struct DatasetPushStatuses {
+    pub statuses: Vec<DatasetPushStatus>,
+}
+
+impl From<kamu_core::services::push_service::DatasetPushStatuses> for DatasetPushStatuses {
+    fn from(value: kamu_core::services::push_service::DatasetPushStatuses) -> Self {
+        let statuses: Vec<DatasetPushStatus> = value
+            .statuses
+            .into_iter()
+            .map(
+                |kamu_core::PushStatus {
+                     remote,
+                     check_result,
+                 }| DatasetPushStatus {
+                    remote: remote.into(),
+                    status: check_result.ok().map(Into::into),
+                },
+            )
+            .collect();
+        Self { statuses }
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// CompareChainsResult
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Enum, Serialize, Debug, Copy, Clone, PartialEq, Eq)]
+pub enum CompareChainsResult {
+    Equal,
+    LhsAhead,
+    LhsBehind,
+    Divergence,
+}
+
+impl From<comp::CompareChainsResult> for CompareChainsResult {
+    fn from(value: comp::CompareChainsResult) -> Self {
+        match value {
+            comp::CompareChainsResult::Equal => CompareChainsResult::Equal,
+            comp::CompareChainsResult::LhsAhead { .. } => CompareChainsResult::LhsAhead,
+            comp::CompareChainsResult::LhsBehind { .. } => CompareChainsResult::LhsBehind,
+            comp::CompareChainsResult::Divergence { .. } => CompareChainsResult::Divergence,
+        }
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// DatasetRefRemote
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct DatasetRefRemote(odf::DatasetRefRemote);
+
+impl From<odf::DatasetRefRemote> for DatasetRefRemote {
+    fn from(value: odf::DatasetRefRemote) -> Self {
+        DatasetRefRemote(value)
+    }
+}
+
```

Comment:
This struct naturally belongs to `./src/adapter/graphql/src/scalars/dataset_id_name.rs`

Reply:
done